### PR TITLE
[Improve][Jdbc] Support postgresql inet type.

### LINKED ISSUE #7817: PostgresSQL and PostgresSQL CDC connector does not support the “inet” data type
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

PostgresSQL and PostgresSQL CDC connectors support inet data  type 
  

### Usage Scenario

  I encountered an error when reading PostgresSQL table data using the PostgresSQL CDC connector. 
  There is an inet type field in my table,and the error message reads: ['Postgres' tables unsupported get catalog table, the corresponding field types in the following tables are not supported: '{"nsc. public. audit": {"ip": "inet"}']. 
  Upon checking the source code, I found that the PostgresTypeConverter class does not support the data type "inet".

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by CosmosNi**
close #7817 
Support postgresql inet type for source and sink

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java [hunk 1]
```diff
@@ -41,5 +41,6 @@ public enum SqlType {
     BFLOAT16_VECTOR,
     SPARSE_FLOAT_VECTOR,
     ROW,
-    MULTIPLE_ROW;
+    MULTIPLE_ROW,
+    INET;
```
Comment (corgy-w) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java]: It seems to me that there is no need for `INET`, which can be transferred in `string` with ST

Comment (hailin0) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java]: +1

Comment (hailin0) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java]: cc @Hisoka-X

Comment (hailin0) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java]: Seatunnel built-in data types require a high degree of abstraction

Comment (CosmosNi) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java]: @hailin0 @corgy-w When  for sink，pg table  also uses the inet type，this cannot be written.But I ignore that the sink is of string type

Comment (CosmosNi) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java:45]: and for source，When the schema is automatically obtained，PostgresTypeConverter class does not support the data type "inet"，so an error was reported before transform

Comment (CosmosNi) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java]: and for source,when  the schema is automatically obtained， PostgresTypeConverter class does not support the data type "inet"，so an error was reported before transform.
image

Comment (corgy-w) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java]: We can increase the pg base type, but we don't need to increase the ST base type @CosmosNi 
image

Comment (CosmosNi) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java]: @corgy-w your idea is good.I'll see if I can solve this problem through configuration.

Comment (Hisoka-X) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SqlType.java]: > It seems to me that there is no need for INET, which can be transferred in string with ST

+1

**Comment by CosmosNi** [comment]
Consider whether to re-obtain the TableSchema of the sink table. Then overwrite AbstractJdbcRowConverter toExternal()  to handle special types.
image

**Comment by CosmosNi** [comment]
@hailin0 @Hisoka-X @corgy-w please cc. support  inet ->inet,inet->string.  Not support string -> inet(customize insert sql to resolve )

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @CosmosNi

**Review by hailin0 — APPROVED** [review]

**Comment by hailin0** [comment]
close #7817

**Pull request merged**